src/wallet/wallet.cpp

```diff
@@ -2777,7 +2777,7 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
                     strFailReason = _("Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.").translated;
                 }
                 scriptChange = GetScriptForDestination(dest);
-                assert(!dest.empty() || scriptChange.empty());
+                CHECK_NONFATAL(!boost::get<CNoDestination>(&dest) == !scriptChange.empty());
```

Comment:
> Could the boost::get() return value be converted to bool explicitly?

Yes, for example with the `!!` "operator".

```suggestion
                CHECK_NONFATAL(!!boost::get<CNoDestination>(&dest) == scriptChange.empty());
```